cidc_api/resources/upload_jobs.py

```diff
@@ -165,6 +177,40 @@ def update_upload_job(upload_job: UploadJobs, upload_job_updates: UploadJobs):
     return upload_job
 
 
+def _remove_optional_uuid_recursive(target: dict, uuid: str):
+    """
+    If target contains an item : dict with {"upload_placeholder":uuid}, removes that item and returns the modified target
+    If no such item is found, continues recursively as depth first search
+    If the uuid is never found, returns the target unchanged
+    """
+    if isinstance(target, dict):
+        if "upload_placeholder" in target and target["upload_placeholder"] == uuid:
```

Comment:
This can be shortened to `target.get("upload_placeholder") == uuid`.

Reply:
in next commit